docs/proposals/error-handling-guideline.md

```diff
@@ -0,0 +1,225 @@
+
+```
+$ oras cp
+Error: accepts 2 arg(s), received 0
+```
+
+Suggested error message:
+
+```
+$ oras cp
+Error: "oras copy" requires exactly 2 arguments.
+
+Usage: oras copy [flags] <from>{:<tag>|@<digest>} <to>[:<tag>[,<tag>][...]]
+Help: Copy artifacts from one target to another. Run "oras copy -h" for more options and examples
+```
+
+#### Example 2: when reference does not match with the expected format
+
+Current behavior and output:
+
+```
+$ oras tag list ghcr.io/oras-project/oras
+Error: unable to add tag for 'list': invalid reference: missing repository
+```
+
+Suggested error message:
+
+```
+$ oras tag list ghcr.io/oras-project/oras
+Error: unable to add tag for 'list': invalid reference: missing repository
+
+Usage: oras tag [flags] <name>{:<tag>|@<digest>} <new_tag> [...]
+Help: Tag a manifest in a registry or an OCI image layout. Run "oras tag -h" for more options and examples
+```
+
+#### Example 3: When fetching a manifest if no manifest tag or digest is provided
+
+Current behavior and output:
+
+```
+$ oras manifest fetch --oci-layout /tmp/ginkgo1163328512 >>
+Error: "/tmp/ginkgo1163328512": no tag or digest when expecting <name:tag|name@digest>
+```
+
+Suggested error message:
+
+```
+$ oras manifest fetch --oci-layout /tmp/ginkgo1163328512 >>
+Error: "/tmp/ginkgo1163328512": no tag or digest specified
+
+Usage: oras manifest fetch [flags] <name>{:<tag>|@<digest>}
+Help: Fetch manifest of the target artifact. Run "oras manifest fetch -h" for more options and examples 
+```
+
+#### Example 4: push a manifest if no media type flag provided
+
+Current behavior and output:
+
+```
+$ oras manifest push --oci-layout /tmp/ginkgo2167255592:mediatype-flag
+Error: media type is not recognized. 
+```
+
+Suggested error message:
+
+```
+$ oras manifest push --oci-layout /tmp/ginkgo2167255592:mediatype-flag
+Error: media type is not recognized. Specify an valid media type with "--media-type"
+```
+
+#### Example 5: attach an artifact if the given option is unknown
+
+Current behavior and output:
+
+```
+$ oras attach --artifact-type oras/test localhost:7000/command/images:foobar --distribution-spec ???
+Error: unknown distribution specification flag: "???"
+```
+
+Suggested error message:
+
+```
+$ oras attach --artifact-type oras/test localhost:7000/command/images:foobar --distribution-spec ???
+Error: unknown distribution specification flag: "???". Available options: v1.1-referrers-api, v1.1-referrers-tag
+```
+
+#### Example 6: when attaching an file, if no file reference or manifest annotation provided
+
+Current behavior and output:
+
+```
+$ oras attach --artifact-type oras/test /tmp/ginkgo2977244222:foobar >>
+Error: no blob or manifest annotation are provided
+```
+
+Suggested error message:
+
+```
+$ oras attach --artifact-type oras/test /tmp/ginkgo2977244222:foobar >>
+Error: no blob or manifest annotation are provided
+
+Usage: oras attach [flags] --artifact-type=<type> <name>{:<tag>|@<digest>} <file>[:<type>] [...]
```

Comment:
The usage is too broad, and not helpful enough, maybe we can show an example for attaching files, and run `oras attach` for more usage and more examples.

Reply:
This got resolved without any comments.

Reply:
Sorry for missing reply to this comment. 

This usage is for general reference if users don't follow the standard form of the command usage. It was defined in the [recommended error message structure](https://github.com/oras-project/oras/pull/1163/files#diff-87978b235bb041f07bf2e71966c261dd6103d718e95ebb9c5c984bebbaf5d357R94).

The detailed prompt message is showed at [line #166](![image](https://github.com/oras-project/oras/pull/1163/files#diff-87978b235bb041f07bf2e71966c261dd6103d718e95ebb9c5c984bebbaf5d357R166)